manager/orchestrator/global/global_test.go

```diff
@@ -524,6 +524,69 @@ func TestInitializationExtraTask(t *testing.T) {
 	assert.Equal(t, liveCnt, 1)
 }
 
+func TestInitializationMultipleServices(t *testing.T) {
+	ctx := context.Background()
+	s := store.NewMemoryStore(nil)
+	assert.NotNil(t, s)
+	defer s.Close()
+
+	// create nodes, services and tasks in store directly
+	// where orchestrator runs, it should fix tasks to declarative state
+	addNode(t, s, node1)
+	addService(t, s, service1)
+	addService(t, s, service2)
+	tasks := []*api.Task{
+		// nodeid1 has 1 task for serviceid1 and 1 task for serviceid2
+		{
+			ID:           "task1",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: service1.Spec.Task,
+			ServiceAnnotations: api.Annotations{
+				Name: "task1",
+			},
+			ServiceID: "serviceid1",
+			NodeID:    "nodeid1",
+		},
+		{
+			ID:           "task2",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: service2.Spec.Task,
+			ServiceAnnotations: api.Annotations{
+				Name: "task2",
+			},
+			ServiceID: "serviceid2",
+			NodeID:    "nodeid1",
+		},
+	}
+	for _, task := range tasks {
+		addTask(t, s, task)
+	}
+
+	// watch orchestration events
+	watch, cancel := state.Watch(s.WatchQueue(), api.EventCreateTask{}, api.EventUpdateTask{}, api.EventDeleteTask{})
+	defer cancel()
+
+	orchestrator := NewGlobalOrchestrator(s)
+	defer orchestrator.Stop()
+
+	go func() {
+		assert.NoError(t, orchestrator.Run(ctx))
+	}()
+
+	// Nothing should happen because both tasks are up to date.
```

Comment:
Yes. Or service1 has a failing task. Make sure reconciliation only touch service1, not service2 tasks.